debezium-core/src/main/java/io/debezium/transforms/TimezoneConverter.java

```diff
@@ -0,0 +1,502 @@
+                throw new DebeziumException(
+                        "Invalid exclude list format. Please specify a list of rules in the format of \"source:<tablename>:<fieldnames>\", \"topic:<topicname>:<fieldnames>\", \"<matchname>:<fieldnames>\"");
+            }
+        }
+
+        if (!validateTimezoneString()) {
+            throw new DebeziumException(
+                    "Invalid timezone format. Please specify either a geographic timezone (e.g. America/Los_Angeles) or a UTC offset in the format of +/-hh:mm, (e.g. +08:00).");
+        }
+
+        if (!includeList.isEmpty() && !excludeList.isEmpty()) {
+            throw new DebeziumException("Both include and exclude lists are specified. Please specify only one.");
+        }
+    }
+
+    private boolean validateTimezoneString() {
+        if (TIMEZONE_PATTERN.matcher(convertedTimezone).matches()) {
+            return true;
+        }
+        else if (ZoneId.getAvailableZoneIds().contains(convertedTimezone)) {
+            return true;
+        }
+        else {
+            return Arrays.asList(TimeZone.getAvailableIDs()).contains(convertedTimezone);
+        }
+    }
+
+    @Override
+    public void close() {
+    }
+
+    private enum Type {
+        ALL,
+        INCLUDE,
+        EXCLUDE
+    }
+
+    private Object getTimestampWithTimezone(String schemaName, Object fieldValue) {
+        Object updatedFieldValue = fieldValue;
+        ZoneId zoneId = ZoneId.of(convertedTimezone);
+        ZoneOffset zoneOffset = zoneId.getRules().getOffset(Instant.now());
+        switch (schemaName) {
+            case ZonedTimestamp.SCHEMA_NAME:
+                OffsetDateTime offsetDateTime = OffsetDateTime.parse((String) fieldValue, DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+                zoneOffset = zoneId.getRules().getOffset(offsetDateTime.toLocalDateTime());
+                OffsetDateTime offsetDateTimeWithZone = offsetDateTime.withOffsetSameInstant(zoneOffset);
+                updatedFieldValue = ZonedTimestamp.toIsoString(offsetDateTimeWithZone, null);
+                break;
+            case ZonedTime.SCHEMA_NAME:
+                OffsetTime offsetTime = OffsetTime.parse((String) fieldValue, DateTimeFormatter.ISO_OFFSET_TIME);
+                OffsetTime offsetTimeWithZone = offsetTime.withOffsetSameInstant(zoneOffset);
+                updatedFieldValue = ZonedTime.toIsoString(offsetTimeWithZone, null);
+                break;
+            case MicroTimestamp.SCHEMA_NAME:
+                long microTimestamp = (long) fieldValue;
+                Instant microInstant = Instant.ofEpochSecond(microTimestamp / 1_000_000, (microTimestamp % 1_000_000) * 1_000);
+                LocalDateTime microLocalDateTime = microInstant.atOffset(ZoneOffset.UTC).toLocalDateTime();
+                zoneOffset = zoneId.getRules().getOffset(microLocalDateTime);
+                updatedFieldValue = microLocalDateTime.atOffset(zoneOffset).toInstant().toEpochMilli() * 1_000;
```

Comment:
@ani-sha You should use combination of `getEpochSecond` and `getNano` otherwise you are losing precision.